src/eth_provider/provider.rs

```diff
@@ -138,11 +139,105 @@ pub struct EthDataProvider<SP: starknet::providers::Provider> {
+
+                // Continue to the next iteration of the loop
+                continue;
+            }
+
+            // Extract the signature from the transaction
+            let transaction_signature = tx.tx.signature.unwrap();
+
+            // Create a signed transaction object from the pending transaction
+            let transaction_signed = TransactionSigned::from_transaction_and_signature(
+                rpc_to_primitive_transaction(tx.tx.clone())?,
+                reth_primitives::Signature {
+                    r: transaction_signature.r,
+                    s: transaction_signature.s,
+                    odd_y_parity: transaction_signature.y_parity.unwrap().0,
+                },
+            );
+
+            // Recover the signer address from the signed transaction
+            let signer = transaction_signed.recover_signer().unwrap();
+
+            // Determine the maximum fee for the transaction
+            let max_fee = {
+                let eth_fees_per_gas =
+                    transaction_signed.effective_gas_price(Some(transaction_signed.max_fee_per_gas() as u64)) as u64;
+                let eth_fees = eth_fees_per_gas.saturating_mul(transaction_signed.gas_limit());
+                let balance = self.balance(signer, None).await?;
+                let max_fee: u64 = balance.try_into().unwrap_or(u64::MAX);
+                max_fee.saturating_sub(eth_fees)
+            };
+
+            // Convert the transaction to a Starknet transaction
+            let starknet_transaction =
+                to_starknet_transaction(&transaction_signed, tx.tx.chain_id.unwrap(), signer, max_fee)?;
+
+            // Add the transaction to the Starknet provider
+            let res = self
+                .starknet_provider
+                .add_invoke_transaction(starknet_transaction)
+                .await
+                .map_err(KakarotError::from)?;
+
+            // Update the pending transaction in the database with the new number of retries
+            self.database
+                .update_one::<StoredPendingTransaction>(
+                    StoredPendingTransaction::new(tx.clone().tx, retries),
+                    hash_filter,
+                    true,
+                )
+                .await?;
+
+            // Return transaction hash if testing feature is enabled, otherwise log and return Ethereum hash
+            if cfg!(feature = "testing") {
+                transactions_retried.push(B256::from_slice(&res.transaction_hash.to_bytes_be()[..]));
+            } else {
+                let hash = transaction_signed.hash();
+                tracing::info!(
+                    "Fired a transaction: Starknet Hash: {:?} --- Ethereum Hash: {:?}",
+                    res.transaction_hash,
+                    hash
+                );
+
+                transactions_retried.push(hash);
+            }
+        }
+
+        // Return the hashes of retried transactions
+        Ok(transactions_retried)
```

Comment:
could we rlp encode the tx and use `self.send_raw_transaction`?

Reply:
Yes partially but not fully because in pending database we store `Transaction` and to be able to call `send_raw_transaction` with `Bytes` we first have to convert this into a `TransactionSigned`. Just did it like basically:
- We catch the transaction from the database pending collection
- We convert to `TransactionSigned`
- We encode
- We call `send_raw_transaction`